test_plans/deduction_guides.asciidoc

```diff
@@ -0,0 +1,111 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for deduction guides
+
+This is a test plan for deduction guides functionality of classes with templates, such as buffer, multi_ptr, range, id and vec.
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line if not stated otherwise.
+
+=== Types coverage
+
+`buffer`, `milti_ptr` and `vec` are tested using types: `char, unsigned char, short, unsigned short, int, unsigned int, long, unsigned long, float.
```

Comment:
```suggestion
`buffer`, `multi_ptr` and `vec` are tested using types: `char`, `unsigned char`, `short`, `unsigned short`, `int`, `unsigned int`, `long`, `unsigned long`, `float`.
```

I'm not opposed to testing with so many types, but I think this might be overkill.  I think it would be OK to cut this down to just a couple if you wanted.

Reply:
I want to integrate deduction guides tests to existing tests, so it would be better to use same type packs.